index.js

```diff
@@ -42,22 +56,38 @@ function walkSync (baseDir, _options, _relativePath) {
   var entries = fs.readdirSync(baseDir + '/' + relativePath).sort();
   for (var i = 0; i < entries.length; i++) {
     var entryRelativePath = relativePath + entries[i];
-    var stats = getStat(baseDir + '/' + entryRelativePath);
+    var fullPath = baseDir + '/' + entryRelativePath;
+    var stats = getStat(fullPath);
 
     if (stats && stats.isDirectory()) {
       if (options.directories !== false && (!m || m.match(entryRelativePath))) {
-        results.push(entryRelativePath + '/');
+        results.push(new Entry(relativePath, stats.mode, stats.size, stats.mtime.getTime()));
       }
-      results = results.concat(walkSync(baseDir, options, entryRelativePath));
+      results = results.concat(_walkSync(baseDir, options, entryRelativePath));
     } else {
       if (!m || m.match(entryRelativePath)) {
-        results.push(entryRelativePath);
+        results.push(new Entry(relativePath, stats.mode, stats.size, stats.mtime.getTime()));
       }
     }
   }
   return results;
 }
 
+function Entry(relativePath, mode, size, mtime) {
+  this.relativePath = relativePath;
+  this.mode = mode;
+  this.size = size;
+  this.mtime = mtime;
+}
+
+Entry.prototype.stats = function () {
+  return {
+    mode: this.mode,
+    size: this.size,
+    mtime: this.mtime
+  };
+}
```

Comment:
I wonder if we should just be honest that we don't have a full stats object and let people use `entry.mtime` etc. directly, rather than giving them "half a stats object"? (There's also the issue that `entry.stats().mtime.getTime()` won't work like on a real stats object.)